tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -262,6 +381,86 @@ static void test_tx_cb_multi(void)
 	PASS();
 }
 
+static void test_tx_proxy_mixin(void)
+{
+	static struct bt_mesh_prov prov = {
+		.uuid = test_prov_uuid,
+	};
+	int err;
+
+	/* Initialize mesh stack and enable pb gatt bearer to emit beacons. */
+	bt_mesh_device_setup(&prov, &comp);
+	err = bt_mesh_prov_enable(BT_MESH_PROV_GATT);
+	ASSERT_OK(err, "Failed to enable GATT provisioner");
+
+	/* Start pb gatt beacons with 100 msec interval */
+	k_sleep(K_MSEC(1800));
+
+	LOG_INF("Provision device under test");
+	/* Provision dut and start gatt proxy beacons. */
+	bt_mesh_provision(test_net_key, 0, 0, 0, adv_cfg.addr, adv_cfg.dev_key);
+
+	/* Start proxy beacons with 1 sec interval */
+	k_sleep(K_MSEC(6000));
+
+	/* Send adv data */
+	struct net_buf *buf = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_LOCAL_ADV,
+			BT_MESH_TRANSMIT(5, 20), K_NO_WAIT);
+	net_buf_add_mem(buf, txt_msg, sizeof(txt_msg));
+	bt_mesh_adv_send(buf, NULL, NULL);
+	k_sleep(K_MSEC(150));
+
+	/* Start proxy beacons again */
+	k_sleep(K_MSEC(6000));
+
+	PASS();
+}
+
+static void test_rx_proxy_mixin(void)
+{
+	static struct bt_mesh_test_gatt gattparam = {
+		.transmits = 1500 / 100,
+		.interval = 100,
+		.tolerance = 20,
+		.service = MESH_PROV_SERVICE
+	};
+
+	bt_init();
+
+	/* Scan pb gatt beacons. */
+	gatt_param = &gattparam;
+	rx_gatt_beacons();
+
+	/* Delay to provision dut */
+	k_sleep(K_MSEC(1000));
+
+	/* Scan proxy beacons. */
+	gattparam.transmits = 5000 / 1000;
+	gattparam.interval = 1000;
+	gattparam.tolerance = 100;
```

Comment:
with growing of interval transmit tolerance is decreased more than twice

Reply:
This is log of test with minor modification. I added logging of adv type in scanner. Scanner works as sniffer in this test. As you can see there is no another advertiser activity. Tolerance is the same as for adv bearer. I'll change in the test. Nevertheless, you can see that precision for pb-gatt advertising is higher than for proxy beaconing or for advertiser bearer.
```
Starting adv_tx_proxy_mixin as device #0
Starting adv_rx_proxy_mixin as device #1
Starting phy with 2 devices
d_01: @00:00:00.000000  *** Booting Zephyr OS build zephyr-v3.1.0-5224-ge2f00abe9248  ***
d_00: @00:00:00.000000  *** Booting Zephyr OS build zephyr-v3.1.0-5224-ge2f00abe9248  ***
d_01: @00:00:00.005768  [00:00:00.005,767] <inf> bt_hci_core: HW Platform: Nordic Semiconductor (0x0002)
d_01: @00:00:00.005768  [00:00:00.005,767] <inf> bt_hci_core: HW Variant: nRF52x (0x0002)
d_01: @00:00:00.005768  [00:00:00.005,767] <inf> bt_hci_core: Firmware: Standard Bluetooth controller (0x00) Version 3.2 Build 0
d_01: @00:00:00.005768  [00:00:00.005,767] <wrn> bt_id: No static addresses stored in controller
d_01: @00:00:00.006488  [00:00:00.006,469] <inf> bt_hci_core: Identity: FF:21:3D:E5:B4:CC (random)
d_01: @00:00:00.006488  [00:00:00.006,469] <inf> bt_hci_core: HCI: version 5.3 (0x0c) revision 0x0000, manufacturer 0x0059
d_01: @00:00:00.006488  [00:00:00.006,469] <inf> bt_hci_core: LMP: version 5.3 (0x0c) subver 0xffff
d_01: @00:00:00.006488  [00:00:00.006,469] <inf> test_adv: Bluetooth initialized
d_00: @00:00:00.005768  [00:00:00.005,767] <inf> bt_hci_core: HW Platform: Nordic Semiconductor (0x0002)
d_00: @00:00:00.005768  [00:00:00.005,767] <inf> bt_hci_core: HW Variant: nRF52x (0x0002)
d_00: @00:00:00.005768  [00:00:00.005,767] <inf> bt_hci_core: Firmware: Standard Bluetooth controller (0x00) Version 3.2 Build 0
d_00: @00:00:00.005768  [00:00:00.005,767] <wrn> bt_id: No static addresses stored in controller
d_00: @00:00:00.006488  [00:00:00.006,469] <inf> bt_hci_core: Identity: C5:AD:ED:38:00:B2 (random)
d_00: @00:00:00.006488  [00:00:00.006,469] <inf> bt_hci_core: HCI: version 5.3 (0x0c) revision 0x0000, manufacturer 0x0059
d_00: @00:00:00.006488  [00:00:00.006,469] <inf> bt_hci_core: LMP: version 5.3 (0x0c) subver 0xffff
d_00: @00:00:00.006488  [00:00:00.006,469] <inf> mesh_test: Bluetooth initialized
d_00: @00:00:00.006488  [00:00:00.006,469] <inf> mesh_test: Mesh initialized
d_00: @00:00:00.006488  [00:00:00.006,469] <inf> bt_mesh_prov_device: Device UUID: 00000000-0000-0000-0000-aa61676e696c
d_01: @00:00:00.023147  [00:00:00.023,132] <inf> test_adv: rx: adv type 0
d_01: @00:00:00.023147  [00:00:00.023,132] <inf> test_adv: rx: adv test +0 ms
d_01: @00:00:00.127364  [00:00:00.127,349] <inf> test_adv: rx: adv type 0
d_01: @00:00:00.127364  [00:00:00.127,349] <inf> test_adv: rx: adv test +107 ms
d_01: @00:00:00.234633  [00:00:00.234,619] <inf> test_adv: rx: adv type 0
d_01: @00:00:00.234633  [00:00:00.234,619] <inf> test_adv: rx: adv test +110 ms
d_01: @00:00:00.344344  [00:00:00.344,329] <inf> test_adv: rx: adv type 0
d_01: @00:00:00.344344  [00:00:00.344,329] <inf> test_adv: rx: adv test +112 ms
d_01: @00:00:00.449202  [00:00:00.449,188] <inf> test_adv: rx: adv type 0
d_01: @00:00:00.449202  [00:00:00.449,188] <inf> test_adv: rx: adv test +107 ms
d_01: @00:00:00.553725  [00:00:00.553,710] <inf> test_adv: rx: adv type 0
d_01: @00:00:00.553725  [00:00:00.553,710] <inf> test_adv: rx: adv test +108 ms
d_01: @00:00:00.654098  [00:00:00.654,083] <inf> test_adv: rx: adv type 0
d_01: @00:00:00.654098  [00:00:00.654,083] <inf> test_adv: rx: adv test +102 ms
d_01: @00:00:00.763930  [00:00:00.763,916] <inf> test_adv: rx: adv type 0
d_01: @00:00:00.763930  [00:00:00.763,916] <inf> test_adv: rx: adv test +113 ms
d_01: @00:00:00.870772  [00:00:00.870,758] <inf> test_adv: rx: adv type 0
d_01: @00:00:00.870772  [00:00:00.870,758] <inf> test_adv: rx: adv test +109 ms
d_01: @00:00:00.977828  [00:00:00.977,813] <inf> test_adv: rx: adv type 0
d_01: @00:00:00.977828  [00:00:00.977,813] <inf> test_adv: rx: adv test +110 ms
d_01: @00:00:01.087228  [00:00:01.087,219] <inf> test_adv: rx: adv type 0
d_01: @00:00:01.087228  [00:00:01.087,219] <inf> test_adv: rx: adv test +112 ms
d_01: @00:00:01.194223  [00:00:01.194,213] <inf> test_adv: rx: adv type 0
d_01: @00:00:01.194223  [00:00:01.194,213] <inf> test_adv: rx: adv test +109 ms
d_01: @00:00:01.299295  [00:00:01.299,285] <inf> test_adv: rx: adv type 0
d_01: @00:00:01.299295  [00:00:01.299,285] <inf> test_adv: rx: adv test +108 ms
d_01: @00:00:01.400613  [00:00:01.400,604] <inf> test_adv: rx: adv type 0
d_01: @00:00:01.400613  [00:00:01.400,604] <inf> test_adv: rx: adv test +104 ms
d_01: @00:00:01.506326  [00:00:01.506,317] <inf> test_adv: rx: adv type 0
d_01: @00:00:01.506326  [00:00:01.506,317] <inf> test_adv: rx: adv test +108 ms
d_01: @00:00:01.506326  [00:00:01.506,317] <inf> test_adv: rx completed. stop observer.
d_00: @00:00:01.765625  [00:00:01.765,625] <inf> test_adv: Provision device under test
d_00: @00:00:01.765625  [00:00:01.765,625] <inf> bt_mesh_main: Primary Element: 0x0001
d_00: @00:00:01.765625  [00:00:01.765,625] <dbg> bt_mesh_main: bt_mesh_provision: net_idx 0x0000 flags 0x00 iv_index 0x0000
d_01: @00:00:02.769138  [00:00:02.769,134] <inf> test_adv: rx: adv type 0
d_01: @00:00:02.769138  [00:00:02.769,134] <inf> test_adv: rx: adv test +0 ms
d_01: @00:00:03.771777  [00:00:03.771,759] <inf> test_adv: rx: adv type 0
d_01: @00:00:03.771777  [00:00:03.771,759] <inf> test_adv: rx: adv test +1027 ms
d_01: @00:00:04.778169  [00:00:04.778,167] <inf> test_adv: rx: adv type 0
d_01: @00:00:04.778169  [00:00:04.778,167] <inf> test_adv: rx: adv test +1030 ms
d_01: @00:00:05.779514  [00:00:05.779,510] <inf> test_adv: rx: adv type 0
d_01: @00:00:05.779514  [00:00:05.779,510] <inf> test_adv: rx: adv test +1026 ms
d_01: @00:00:06.787524  [00:00:06.787,506] <inf> test_adv: rx: adv type 0
d_01: @00:00:06.787524  [00:00:06.787,506] <inf> test_adv: rx: adv test +1032 ms
d_01: @00:00:06.787524  [00:00:06.787,506] <inf> test_adv: rx completed. stop observer.
d_01: @00:00:07.629173  [00:00:07.629,150] <inf> test_adv: rx: adv test +0 ms
d_01: @00:00:07.659141  [00:00:07.659,118] <inf> test_adv: rx: adv test +30 ms
d_01: @00:00:07.685936  [00:00:07.685,913] <inf> test_adv: rx: adv test +28 ms
d_01: @00:00:07.705955  [00:00:07.705,932] <inf> test_adv: rx: adv test +20 ms
d_01: @00:00:07.734215  [00:00:07.734,191] <inf> test_adv: rx: adv test +29 ms
d_01: @00:00:07.763939  [00:00:07.763,916] <inf> test_adv: rx: adv test +31 ms
d_01: @00:00:07.763939  [00:00:07.763,916] <inf> test_adv: rx completed. stop observer.
d_01: @00:00:07.786746  [00:00:07.786,743] <inf> test_adv: rx: adv type 0
d_01: @00:00:07.786746  [00:00:07.786,743] <inf> test_adv: rx: adv test +0 ms
d_01: @00:00:08.791125  [00:00:08.791,107] <inf> test_adv: rx: adv type 0
d_01: @00:00:08.791125  [00:00:08.791,107] <inf> test_adv: rx: adv test +1029 ms
d_01: @00:00:09.795654  [00:00:09.795,623] <inf> test_adv: rx: adv type 0
d_01: @00:00:09.795654  [00:00:09.795,623] <inf> test_adv: rx: adv test +1028 ms
d_01: @00:00:10.795536  [00:00:10.795,532] <inf> test_adv: rx: adv type 0
d_01: @00:00:10.795536  [00:00:10.795,532] <inf> test_adv: rx: adv test +1024 ms
d_01: @00:00:10.795536  [00:00:10.795,532] <inf> test_adv: rx completed. stop observer.
d_01: @00:00:10.795537 INFO: test_rx_proxy_mixin PASSED
d_00: @00:00:13.633790 INFO: test_tx_proxy_mixin PASSED
```

Reply:
It looks like a rounding issue, because if set `CONFIG_SYS_CLOCK_TICKS_PER_SEC=32768` the so called tolerance will be within 10ms which is closer to the truth. Maybe interval + 15ms is reasonable fluctuation for this kind of test. Delta for interval check could I think also be updated in `xmit_scan_cb`. At least it shouldn't be double of interval.